File: ulyp-ui/src/main/java/com/ulyp/ui/renderers/RenderedBoolean.java
package com.ulyp.ui.renderers;

import com.ulyp.core.printers.BooleanRepresentation;
import com.ulyp.ui.RenderSettings;
import com.ulyp.ui.util.StyledText;

import static com.ulyp.ui.util.CssClass.*;

public class RenderedBoolean extends RenderedObject {

    protected RenderedBoolean(BooleanRepresentation representation, RenderSettings renderSettings) {
        super(representation.getType());

        if (renderSettings.showTypes()) {
            super.getChildren().add(StyledText.of(representation.getType().getName(), CALL_TREE_TYPE_NAME));
            super.getChildren().add(StyledText.of(": ", CALL_TREE_PLAIN_TEXT));
        }
        super.getChildren().add(StyledText.of(String.valueOf(representation.value()), CALL_TREE_NUMBER));
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/renderers/RenderedBoolean.kt
package com.ulyp.ui.renderers

import com.ulyp.core.recorders.BooleanRecord
import com.ulyp.ui.RenderSettings
import com.ulyp.ui.util.CssClass
import com.ulyp.ui.util.StyledText.of

class RenderedBoolean(representation: BooleanRecord, renderSettings: RenderSettings) : RenderedObject(representation.type) {

    init {
        if (renderSettings.showTypes()) {
            super.getChildren().add(of(representation.type.name, CssClass.CALL_TREE_TYPE_NAME))
            super.getChildren().add(of(": ", CssClass.CALL_TREE_NODE_SEPARATOR))
        }
        super.getChildren().add(of(representation.value().toString(), CssClass.CALL_TREE_NUMBER))
    }
}